python/cudf/cudf/core/column/string.py

```diff
@@ -4689,6 +4692,14 @@ def subword_tokenize(
         array([[0, 0, 2],
                [1, 0, 1]], dtype=uint32)
         """
+        warning_message = (
+            "`ser.str.subword_tokenize` API will be depreciated"
+            "in future versions of cudf.\n"
+            "Use `cudf.core.subword_tokenizer.SubwordTokenizer` "
+            "instead"
+        )
+
+        warnings.warn(warning_message, DeprecationWarning, stacklevel=2)
```

Comment:
Added `stacklevel` because by default warnings are filtered out in `jupyter-notebooks` . This is because warnings only show up . I could not find a cleaner way to surface it out to `jupyter` environments. 

Related PEP https://www.python.org/dev/peps/pep-0565/

Reply:
We can follow `pandas` and raise `FutureWarning `. Was not sure what our policy is .  

Example : 
https://github.com/pandas-dev/pandas/blob/3ce150d2594d9738c5e43139194692151be4f2a8/pandas/io/date_converters.py#L15-L22

Reply:
Looks like we don't have `DeprecationWarning` in our code-base, can you try just `warnings.warn(warning_message)` and if that solved the issue?

Reply:
I don't follow what the issue is with `DeprecationWarning` (it's a Python built-in), but one thing to keep in mind is that unlike most warnings `DeprecationWarning` is suppressed by default, so users have to explicitly use the `warnings` library to opt in to seeing them. I don't know if that affects your choice of warning here.

Reply:
>  can you try just warnings.warn(warning_message) and if that solved the issue 

@galipremsagar , We do use `DeprecationWarning `  in `dask-cudf`  (see [link](https://github.com/rapidsai/cudf/blob/99808abc405f63c66bc2fccde521192655c646c0/python/dask_cudf/dask_cudf/core.py#L307-L311)) so I assumed it would be fine to use it here. 

We also use `FutureWarning` (see [example](https://github.com/rapidsai/cudf/blob/d8a3e60c7945707bc45c1a9733477a5d24c1b50f/python/cudf/cudf/core/multiindex.py#L634-L638)) .

I think we should use `FutureWarning`   as it is recommended in the python documentation for deprecated features when warnings are intended for end-users of applications that are written in Python. 

For context libraries like `pandas` and sckit learn both use `FutureWarning` (See [section](https://github.com/scikit-learn/scikit-learn/blob/5a879f4a024d499c74da65f0343b535be4aa096e/doc/developers/contributing.rst#change-the-default-value-of-a-parameter) on contributing.md of `sckit-learn`) 

>DeprecationWarning is suppressed by default

@vyasr , Yup. That is a problem with `DeprecationWarning` and might be there with `FutureWarnings` too when the function is called outside `__main__` . That said we can surface them using `stack level.

Reply:
We can use `FutureWarning` in that case as we want these warnings to be seen by the users. @vyasr What do you suggest?

Reply:
Switched to FutureWarning with commit  https://github.com/rapidsai/cudf/pull/8603/commits/1354201b6018d48e99a68c228537b2ad51db0ecd

Reply:
Yes I agree that `FutureWarning` would be a good choice here. I'm going to resolve this conversation but feel free to reopen if we need more discussion.